[iamdiluxedbutcooler] iP

iamdiluxedbutcooler opened the pull request:
## Puke 🤮 Task Manager

> " I drink 'til I'm drunk (yeah), smoke 'til I'm high (yeah)" —  your friendly bartender ([source](https://www.youtube.com/watch?v=G5wNaoRs8sI))

**Puke** frees your mind of having to remember things you need to do. It's:
- **text-based**
- _easy to learn_
- **~~_FAST_~~ SUPER FAST** to use

### Getting Started with Puke
All you need to do is,
1. download it from [here](https://www.youtube.com/watch?v=dQw4w9WgXcQ).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 🤢

And it is **FREE**!

### Features
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public static void main(String[] args) throws PukeException {
        Puke chatBot = new Puke();
        chatBot.ui.start();
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

iamdiluxedbutcooler committed:
Rename main class

iamdiluxedbutcooler committed:
Add universal message builder class to send messages

iamdiluxedbutcooler committed:
Add greeting and goodbye message

iamdiluxedbutcooler committed:
Add constructor for the chatbot, message obj

iamdiluxedbutcooler committed:
Add new chatbot obj, start method that send greeting, terminate method that send goodbye

iamdiluxedbutcooler committed:
Edit the goodbye message

iamdiluxedbutcooler committed:
Add echo method

iamdiluxedbutcooler committed:
Add "Puke" more personality

iamdiluxedbutcooler committed:
Add echo implementation to the chatbot

iamdiluxedbutcooler committed:
Edit the personality part of message

iamdiluxedbutcooler committed:
Edit the personality part of message

iamdiluxedbutcooler committed:
Edit the personality part of message (again)

iamdiluxedbutcooler committed:
Change from 'echoMessage' to 'sendMessage' for more precise name for universal use

iamdiluxedbutcooler committed:
Add task class

iamdiluxedbutcooler committed:
Add input manager class to handle each type of command

iamdiluxedbutcooler committed:
Edit Puke main class, implement new input manager

iamdiluxedbutcooler committed:
Implement task class

iamdiluxedbutcooler committed:
Edit task manager: change elem type from string to task

iamdiluxedbutcooler committed:
Edit task manager: add marking functionality

iamdiluxedbutcooler committed:
Update input manager to handle new task commands

iamdiluxedbutcooler committed:
Update Task class: add @Override at toString

iamdiluxedbutcooler committed:
Update README.md

iamdiluxedbutcooler committed:
Add Todo class

iamdiluxedbutcooler committed:
Add Deadline class

iamdiluxedbutcooler committed:
Add Event class

iamdiluxedbutcooler committed:
Update TaskManager class, with define the task type functionality

iamdiluxedbutcooler committed:
Update InputManager class to handle task type commands

iamdiluxedbutcooler committed:
Update test to match new class name and directory

iamdiluxedbutcooler committed:
Add input command

iamdiluxedbutcooler committed:
Add expected response

iamdiluxedbutcooler committed:
Add exception class

iamdiluxedbutcooler committed:
Add get method for task counter

iamdiluxedbutcooler committed:
Update input manager to throw exception instead

iamdiluxedbutcooler committed:
Update test input and expected responses for exception cases

iamdiluxedbutcooler committed:
change handleInput in InputManager to switch case approach, making it cleaner

iamdiluxedbutcooler committed:
Add delete method

iamdiluxedbutcooler committed:
Add delete method

iamdiluxedbutcooler committed:
Change to use universal string trimmer

iamdiluxedbutcooler committed:
Update test for delete

iamdiluxedbutcooler committed:
Implement ArrayList

iamdiluxedbutcooler committed:
Add exceptions

iamdiluxedbutcooler committed:
Change exception to the new one

iamdiluxedbutcooler committed:
Add Exceptions classes

iamdiluxedbutcooler committed:
Update InputManager to throw new exception accordingly

iamdiluxedbutcooler committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/InputManager.java
#	src/main/java/exceptions/InvalidTaskNumberFormatException.java
#	src/main/java/exceptions/MissingEventTimeException.java
#	src/main/java/exceptions/MissingTaskNumberException.java
#	src/main/java/exceptions/PukeException.java
#	src/main/java/exceptions/TaskNumberOutOfBoundsException.java

iamdiluxedbutcooler committed:
Change task class to abstract

iamdiluxedbutcooler committed:
Update task classes with toFileFormat method

iamdiluxedbutcooler committed:
Add Storage.java class with save and load method

iamdiluxedbutcooler committed:
Update TaskManager.java to implement save and load data functionality

iamdiluxedbutcooler committed:
Merge branch-Level-7 into master

iamdiluxedbutcooler committed:
Add date time system to Deadline.java

iamdiluxedbutcooler committed:
Add date time system to Event.java

iamdiluxedbutcooler committed:
Merge branch-Level-8 into master

iamdiluxedbutcooler committed:
Make UI related class OOP

iamdiluxedbutcooler committed:
Make Input handler: InputManager OOP

iamdiluxedbutcooler committed:
Make commands OOP

iamdiluxedbutcooler committed:
Refactor to Puke package

iamdiluxedbutcooler committed:
Merge branch-A-MoreOOP into master

iamdiluxedbutcooler committed:
Merge add-gradle-support into master

iamdiluxedbutcooler committed:
Update task manager arguments

iamdiluxedbutcooler committed:
Add TaskManagerTest

iamdiluxedbutcooler committed:
Add InputManagerTest

iamdiluxedbutcooler committed:
Add JavaDocs to exceptions

iamdiluxedbutcooler committed:
Add JavaDocs to commands

iamdiluxedbutcooler committed:
Add JavaDocs to InputManager

iamdiluxedbutcooler committed:
Add JavaDocs to Storage

iamdiluxedbutcooler committed:
Add JavaDocs to tasks

iamdiluxedbutcooler committed:
Add JavaDocs to ui classes

iamdiluxedbutcooler committed:
Add JavaDocs to main class

iamdiluxedbutcooler committed:
Add JavaDocs to tests

iamdiluxedbutcooler committed:
Update formatting

iamdiluxedbutcooler committed:
Add find function matching regex

iamdiluxedbutcooler committed:
Merge branch 'branch-A-JavaDoc'

iamdiluxedbutcooler committed:
Merge branch 'branch-A-CodingStandard'

iamdiluxedbutcooler committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/puke/handlers/InputManager.java

iamdiluxedbutcooler committed:
Delete unnecessary imports

iamdiluxedbutcooler committed:
Add XML styling scripts

iamdiluxedbutcooler committed:
Add checkstyle related dependencies

iamdiluxedbutcooler committed:
Address Checkstyle violations and improve code formatting

iamdiluxedbutcooler committed:
Resolve more code formatting issues

iamdiluxedbutcooler committed:
Merge branch-A-CheckStyle into master

munhuikim commented on src/main/java/puke/Parser.java:
```diff
@@ -0,0 +1,89 @@
+import puke.exceptions.UnknownCommandException;
+import puke.message.MessageBuilder;
+
+/**
```
Great job with the javadocs and naming! It is easy to understand.

munhuikim commented on src/main/java/puke/TaskList.java:
```diff
@@ -0,0 +1,159 @@
+     *
+     * @param loadedFromStorage the tasks loaded from storage
+     */
+    public TaskList(ArrayList<Task> loadedFromStorage) {
```
Nice use of descriptive variable names; it really improves the readability of the code.

munhuikim reviewed (commented):
Overall, LGTM! I think everything covers the coding standards.

thortol commented on src/main/java/puke/task/Event.java:
```diff
@@ -0,0 +1,75 @@
+ */
+public class Event extends Task {
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMM dd yyyy, HH:mm");
+    private static final DateTimeFormatter inputFormatter = DateTimeFormatter.ofPattern("dd/MM/yyyy HHmm");
```
Should constant names be screaming snake case?

thortol commented on src/main/java/puke/task/Task.java:
```diff
@@ -0,0 +1,109 @@
+     *
+     * @return the hash code of the task
+     */
+    @Override
+    public int hashCode() {
+        return Objects.hash(getDescription(), isDone());
+    }
```
Great that you remembered to override hashCode!

thortol reviewed (commented):
Overall LGTM! Couldn't find much errors. Good job!

iamdiluxedbutcooler committed:
Add profile image for GUI

iamdiluxedbutcooler committed:
Add GUI functionality

iamdiluxedbutcooler committed:
Merge branch-Level-10 into master

iamdiluxedbutcooler committed:
Add WrongDateTimeFormatException

iamdiluxedbutcooler committed:
Update Main class in gradle dependencies

iamdiluxedbutcooler committed:
Configure per checkstyle

iamdiluxedbutcooler committed:
Resolve typo error in JavaDocs

iamdiluxedbutcooler committed:
Integrate assert statements across key class methods

Currently, the application lacks comprehensive runtime checks to enforce important invariants and assumptions in the code. This absence can lead to obscure bugs and unexpected behavior in production, making debugging and maintenance more challenging.

Introduce assert statements in constructors and methods across various classes including Task, TaskList, Parser, and Storage to ensure critical conditions such as non-null parameters, valid state transitions, and proper range values are consistently met.

Using assert statements provides a straightforward way to document and enforce these assumptions directly in the code without imposing overhead in production, as assertions can be disabled at runtime if needed.

This approach improves code robustness and aids in early detection of logical errors during the development phase, ultimately enhancing the reliability of the application.

No additional performance overhead is expected in production environments as assertions are typically disabled. This ensures the benefits of enhanced error checking during development while maintaining efficiency in live deployments.

iamdiluxedbutcooler committed:
Fix code quality: header comments issues

iamdiluxedbutcooler committed:
Fix code quality: screaming snake case for static final variable issues

iamdiluxedbutcooler committed:
Fix check style problem

iamdiluxedbutcooler committed:
Merge branch-A-Assertions into master

Integrate assert statements across key class methods

iamdiluxedbutcooler committed:
Merge branch 'master' into branch-A-CodeQuality

iamdiluxedbutcooler committed:
Merge branch-A-CodeQuality into master

Fix code quality issues

iamdiluxedbutcooler committed:
Merge branch-A-CodeQuality into master

Finalize by update branch-A-CodeQuality with updated master

iamdiluxedbutcooler committed:
Remove unnecessary print

iamdiluxedbutcooler committed:
Update parser test

iamdiluxedbutcooler committed:
Add UpdateTaskCommand for C-Update extension

iamdiluxedbutcooler committed:
Add an exception for C-Update extension

iamdiluxedbutcooler committed:
Update existed file to handle C-Update extension

iamdiluxedbutcooler committed:
Merge brance for extension into master

iamdiluxedbutcooler committed:
Add title bar for GUI

iamdiluxedbutcooler committed:
Add product sample UI

iamdiluxedbutcooler committed:
Update user guide

iamdiluxedbutcooler committed:
Remove unnecessary import

iamdiluxedbutcooler committed:
Update user guide

iamdiluxedbutcooler committed:
Update user guide: add link to release

iamdiluxedbutcooler committed:
Update README.md

iamdiluxedbutcooler committed:
Update user guide
